api/krusty/helmchartinflationgenerator_test.go

```diff
@@ -233,3 +236,66 @@ spec:
   type: ClusterIP
 `)
 }
+
+func TestHelmChartInflationGeneratorMultipleValuesFiles(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t)
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	copyValuesFilesTestChartsIntoHarness(t, th)
+
+	th.WriteK(th.GetRoot(), `
+helmCharts:
+  - name: test-chart
+    releaseName: test-chart
+    additionalValuesFiles:
+    - file1.yaml
+    - file2.yaml
+`)
+
+	m := th.Run(th.GetRoot(), th.MakeOptionsPluginsEnabled())
+	asYaml, err := m.AsYaml()
+	require.NoError(t, err)
+	require.Equal(t, string(asYaml), `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    chart: test-1.0.0
+  name: my-deploy
+  namespace: file-2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: test
+  template:
+    spec:
+      containers:
+      - image: test-image-file1:file1
+        imagePullPolicy: Never
+`)
+}
+
+func copyFileIntoHarness(th *kusttest_test.HarnessEnhanced, src, dst string) {
+	bytes, _ := os.ReadFile(src)
+	th.WriteF(dst, string(bytes))
+}
+
+func copyValuesFilesTestChartsIntoHarness(t *testing.T, th *kusttest_test.HarnessEnhanced) {
+	t.Helper()
+
+	thDir := filepath.Join(th.GetRoot(), "charts", "test-chart")
+	chartDir := "./testdata/helmcharts/multiple-values-files/charts/test-chart"
+
+	fs := th.GetFSys()
+	require.NoError(t, fs.MkdirAll(filepath.Join(thDir, "templates")))
+
+	copyFileIntoHarness(th, filepath.Join(chartDir, "Chart.yaml"), filepath.Join(thDir, "Chart.yaml"))
```

Comment:
Is this series essentially copying the full the directory contents, or are we avoiding some files? Fwiw I see kyaml has a `CopyDir`.

Reply:
I changed `CopyDir` to accept an fSys argument and used it. (I also updated localize to use kyaml.CopyDir instead of writing its own)

Reply:
I didn't know about it either fwiw... I just guessed that it might exist (we do so much FS manipulation!), searched for it and got lucky. 😄